firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/FeedbackActivity.java

```diff
@@ -100,14 +103,36 @@ private void setupView() {
     findViewById(R.id.backButton).setOnClickListener(v -> finish());
     findViewById(R.id.sendButton).setOnClickListener(this::submitFeedback);
 
+    // Registers a photo picker activity launcher in single-select mode
+    ActivityResultLauncher<PickVisualMediaRequest> pickMedia =
+        registerForActivityResult(
+            new PickVisualMedia(),
+            uri -> {
+              if (uri != null) {
+                LogWrapper.d(TAG, "Selected custom screenshot URI: " + uri);
+                screenshotUri = uri;
+                setupScreenshot();
+              } else {
+                LogWrapper.d(TAG, "No custom screenshot selected. Not changing screenshot URI.");
+              }
+            });
+
+    findViewById(R.id.chooseScreenshotButton)
+        .setOnClickListener(
+            v ->
+                pickMedia.launch(
+                    new PickVisualMediaRequest.Builder()
+                        .setMediaType(new PickVisualMedia.SingleMimeType("image/png"))
+                        .build()));
+
     setupScreenshot();
   }
 
   private void setupScreenshot() {
     blockingExecutor.execute(
         () -> {
-          // do I/O on separate thread in order to not block the UI
-          Bitmap screenshot = screenshotUri == null ? null : readScreenshot();
+          // Do I/O on separate thread in order to not block the UI
+          Bitmap screenshot = readScreenshot(screenshotUri);
```

Comment:
Now that `screenshotUri` could technically be modified on another thread I don't want to rely on screenshotUri being the same for the entirety of the execution of `readScreenshot()`